services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchSpec.groovy

```diff
@@ -24,4 +33,125 @@ class SwitchSpec extends BaseSpecification {
+        }
+    }
+
+    def "Cannot delete inactive switch with single flow"() {
+        requireProfiles("virtual")
+
+        given: "An active switch with flow"
+        def sw = topology.getActiveSwitches()[0]
+        def flow = flowHelper.singleSwitchFlow(sw)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        when: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(sw.getDpId())
+
+        and: "Try to delete inactive switch with flow"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has flow"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back and remove flow"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundService.allSwitches.any { it.switchId == sw.dpId }
+        }
+        northboundService.deleteFlow(flow.id)
+    }
+
+    def "Cannot delete inactive switch with flow"() {
+        requireProfiles("virtual")
+
+        given: "An active switch with flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.getActiveSwitches()[0..1]
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        when: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(srcSwitch.getDpId())
+
+        and: "Try to delete inactive switch with flow"
+        northboundService.deleteSwitch(srcSwitch.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has flow"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back and remove flow"
+        lockKeeperService.reviveSwitch(srcSwitch.getDpId())
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundService.allSwitches.any { it.switchId == srcSwitch.dpId }
+        }
+        northboundService.deleteFlow(flow.id)
+    }
+
+    def "Can delete active switch if using force delete"() {
+        requireProfiles("virtual")
+
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        when: "Try to force delete active switch with ISLs"
+        def response = northboundService.deleteSwitch(sw.getDpId(), true)
+
+        then: "Check that switch is actually deleted"
+        response.deleted
+        Wrappers.wait(WAIT_OFFSET) {
+            !northboundService.allSwitches.any { it.switchId == sw.dpId }
+        }
+
+        and: "Cleanup: Restore the switch"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundService.allSwitches.any { it.switchId == sw.dpId }
```

Comment:
This still affects subsequent tests. I'm wondering whether you should wait for it to have ACTIVATED status and wait until all related ISLs are DISCOVERED (this may require at most `kilda_discovery_interval` seconds to discover an ISL)

Reply:
okay, lets wait ISLs